iothub_client/inc/iothub_client_core_common.h

```diff
@@ -17,137 +17,138 @@ extern "C"
-    IOTHUB_CLIENT_IOTHUB_METHOD_STATUS_ERROR      \
 
-    /** @brief Enumeration returned by remotely executed functions
-    */
-    MU_DEFINE_ENUM_WITHOUT_INVALID(IOTHUB_CLIENT_IOTHUB_METHOD_STATUS, IOTHUB_CLIENT_IOTHUB_METHOD_STATUS_VALUES);
+typedef enum IOTHUB_IDENTITY_TYPE_TAG
+{
+    IOTHUB_TYPE_TELEMETRY = 0,
+    IOTHUB_TYPE_DEVICE_TWIN = 1,
+    IOTHUB_TYPE_DEVICE_METHODS = 2,
+    IOTHUB_TYPE_EVENT_QUEUE  = 3
+} IOTHUB_IDENTITY_TYPE;
 
+const char* IOTHUB_IDENTITY_TYPEStrings(IOTHUB_IDENTITY_TYPE value);
 
-#define IOTHUB_CLIENT_CONFIRMATION_RESULT_VALUES     \
-    IOTHUB_CLIENT_CONFIRMATION_OK,                   \
-    IOTHUB_CLIENT_CONFIRMATION_BECAUSE_DESTROY,      \
-    IOTHUB_CLIENT_CONFIRMATION_MESSAGE_TIMEOUT,      \
-    IOTHUB_CLIENT_CONFIRMATION_ERROR                 \
 
-    /** @brief Enumeration passed in by the IoT Hub when the event confirmation
-    *           callback is invoked to indicate status of the event processing in
-    *           the hub.
-    */
-    MU_DEFINE_ENUM_WITHOUT_INVALID(IOTHUB_CLIENT_CONFIRMATION_RESULT, IOTHUB_CLIENT_CONFIRMATION_RESULT_VALUES);
+typedef enum IOTHUB_PROCESS_ITEM_RESULT_TAG
+{
+    IOTHUB_PROCESS_OK = 0,
+    IOTHUB_PROCESS_ERROR = 1,
+    IOTHUB_PROCESS_NOT_CONNECTED = 2,
+    IOTHUB_PROCESS_CONTINUE = 3
+} IOTHUB_PROCESS_ITEM_RESULT;
+const char * IOTHUB_PROCESS_ITEM_RESULTStrings(IOTHUB_PROCESS_ITEM_RESULT value);
 
-#define IOTHUB_CLIENT_CONNECTION_STATUS_VALUES             \
-    IOTHUB_CLIENT_CONNECTION_AUTHENTICATED,                \
-    IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED               \
 
+typedef enum IOTHUBMESSAGE_DISPOSITION_RESULT_TAG
+{
+    IOTHUBMESSAGE_ACCEPTED = 0,
+    IOTHUBMESSAGE_REJECTED = 1,
+    IOTHUBMESSAGE_ABANDONED = 2
 
-    /** @brief Enumeration passed in by the IoT Hub when the connection status
-    *           callback is invoked to indicate status of the connection in
-    *           the hub.
-    */
-    MU_DEFINE_ENUM_WITHOUT_INVALID(IOTHUB_CLIENT_CONNECTION_STATUS, IOTHUB_CLIENT_CONNECTION_STATUS_VALUES);
-
-#define IOTHUB_CLIENT_CONNECTION_STATUS_REASON_VALUES      \
-    IOTHUB_CLIENT_CONNECTION_EXPIRED_SAS_TOKEN,            \
-    IOTHUB_CLIENT_CONNECTION_DEVICE_DISABLED,              \
-    IOTHUB_CLIENT_CONNECTION_BAD_CREDENTIAL,               \
-    IOTHUB_CLIENT_CONNECTION_RETRY_EXPIRED,                \
-    IOTHUB_CLIENT_CONNECTION_NO_NETWORK,                   \
-    IOTHUB_CLIENT_CONNECTION_COMMUNICATION_ERROR,          \
-    IOTHUB_CLIENT_CONNECTION_NO_PING_RESPONSE,             \
-    IOTHUB_CLIENT_CONNECTION_OK                            \
-
-    /** @brief Enumeration passed in by the IoT Hub when the connection status
-    *           callback is invoked to indicate status of the connection in
-    *           the hub.
-    */
-    MU_DEFINE_ENUM_WITHOUT_INVALID(IOTHUB_CLIENT_CONNECTION_STATUS_REASON, IOTHUB_CLIENT_CONNECTION_STATUS_REASON_VALUES);
+} IOTHUBMESSAGE_DISPOSITION_RESULT;
```

Comment:
this one doesn't seem to have a `Strings()` function. by design? do we only now put them in if the compiler complains it doesn't exist? Basically only if the `to_string()` call is made for that enum?

Reply:
Bug.  I basically wanted to get PR out for early feedback knowing there were gaps.  I shall address before round 2.

---
In reply to: [350329737](https://github.com/Azure/azure-iot-sdk-c/pull/1342#discussion_r350329737) [](ancestors = 350329737)